gridt/src/app/core/settings.service.ts

```diff
@@ -30,15 +30,13 @@ export class SettingsService {
    * @param identity Settings to be stored.
    */
   public setLocalIdentity$(identity: Identity): Observable<boolean> {
-    return this.auth.isLoggedIn$.pipe(
-      take(1),
-      tap( (val) => {
-        if (val === true) {
-          console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
-          this.secStore.set$("identity", identity).subscribe();
-        }
-      }),
-      mergeMap( val => iif( () => val === true, of(true), throwError("Not logged in: can't store identity in local storage.")))
+    return this.auth.readyAuthentication$.pipe(
+      catchError( () => throwError("Could not set identity: not logged in")),
+      take(1), // Makes sure the observable completes
+      flatMap( () => this.secStore.set$("identity", identity).pipe(
+          catchError(()=> throwError("Could not set identity: secure storage not available."))
+        )
+      )
     )
```

Comment:
There should be a semi colon here and maybe you can use the error code pattern here.

These are the last pieces of correction I'd like to give before accepting and merging the code.

Reply:
I took the liberty to also add some custom errors for the localIdentity$ observable. The error codes are also used in the test, such that we don't have to change them twice.